react-components/src/components/RevealCanvas/RevealCanvas.tsx

```diff
@@ -0,0 +1,26 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+import { type ReactNode, type ReactElement, useRef, useEffect } from 'react';
+import { createPortal } from 'react-dom';
+import { useReveal } from './ViewerContext';
+
+export function RevealCanvas({ children }: { children?: ReactNode }): ReactElement {
+  const viewer = useReveal();
+  const parentElement = useRef<HTMLDivElement>(null);
+
+  return (
+    <div style={{ width: '100%', height: '100%', overflow: 'hidden' }} ref={parentElement}>
+      {mountChildren()}
+    </div>
+  );
+
+  function mountChildren(): ReactElement {
+    useEffect(() => {
+      if (parentElement.current !== null) {
+        parentElement.current.appendChild(viewer.domElement);
+      }
+    }, [parentElement.current]);
```

Comment:
Logically, `parentElement.current` is a mutable value that will not trigger a re-render when it changes, are you sure this gets triggered when `parentElement.current`

Reply:
No, I don't think it will retrigger automatically, but if the useEffect runs again at some point, it can use `parentElement.current` to see whether the call is actually needed

Reply:
Maybe it's not entirely needed, but I don't think it's a problem to keep it